util/progress/progressui/display.go

```diff
@@ -265,6 +274,140 @@ func (d *plainDisplay) done() {
 	d.t.printErrorLogs(d.t.w)
 }
 
+type jsonDisplay struct {
+	t    *trace
+	logs *logMux
+	enc  *json.Encoder
+}
+
+// newJSONDisplay creates a new SolveStatusDisplay that outputs the completed vertex
+// progress indicator for each vertex in JSON format.
+func newJSONDisplay(w io.Writer, opts ...DisplayOpt) Display {
+	enc := json.NewEncoder(w)
+	enc.SetIndent("", "  ")
+	return Display{
+		disp: &jsonDisplay{
+			t:    newTrace(w, false),
+			logs: newLogMux(),
+			enc:  enc,
+		},
+	}
+}
+
+func (d *jsonDisplay) init(displayLimiter *rate.Limiter) {
+	// Display limiter is ignored for the JSON display.
+}
+
+func (d *jsonDisplay) update(ss *client.SolveStatus) {
+	d.t.update(ss, 80)
```

Comment:
Setting `80` in here seems terminal-specific.

Reply:
This number isn't used for non-tty displays. It's mostly an artifact of the previous implementation where `trace` was used for both the tty display and the plain display. The text display used 80 as a default just to give a value (which is unused) and I just carried that on.

Might be worth refactoring this code at some point just to reduce confusion for the implementation but it seemed a bit too large of a task for this PR.

Reply:
If it is unused, then just setting a zero value is ok for now.